src/main.rs

```diff
@@ -21,15 +21,17 @@ pub mod input_editor;
 pub mod peg;
 pub mod expansion;
 
-pub type Variables = BTreeMap<String, String>;
+pub struct Mode {
```

Comment:
Docs would be nice, but the whole `modes` thing is going to get a massive rework eventually, so I am not overly concerned.